esphome/components/sensor/filter.cpp

```diff
@@ -302,11 +308,17 @@ optional<float> DeltaFilter::new_value(float value) {
       return this->last_value_ = value;
     }
   }
-  float diff = fabsf(value - this->last_value_);
-  if (std::isnan(this->last_value_) || (diff > 0.0f && diff >= this->current_delta_)) {
-    if (this->percentage_mode_) {
-      this->current_delta_ = fabsf(value * this->delta_);
-    }
+
+  float min = this->min_percentage_mode_ ? fabsf(value * this->min_delta_) : this->min_delta_;
+  float max = this->min_percentage_mode_ ? fabsf(value * this->max_delta_) : this->max_delta_;
+  float ref = this->baseline_ ? this->baseline_.value()(value) : this->last_value_;
+  float delta = fabsf(value - ref);
+
+  if (std::isnan(ref)) {
+    return this->last_value_ = value;
```

Comment:
Don't use the value of an assignment, it looks too much like a comparison. 
```suggestion
    this->last_value_ = value;
    return value;
```

Reply:
This is what was already here, but definitely agree. Will change